resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -242,12 +243,19 @@ private[spark] class KubernetesSuite extends SparkFunSuite
           action match {
             case Action.ADDED | Action.MODIFIED =>
               execPods(name) = resource
+              // If testing decomissioning delete the node 10 seconds after
+              if (decomissioningTest) {
+                Thread.sleep(1000)
```

Comment:
Why the thread sleep? Why not just check if the `getStatus` is running and then kill?

Reply:
We probably want to wait some fudge factor above running to ensure it has a chance to properly register and everything but yeah we can decrease the fudge factor and check the pod status to be more reliable.